### PR TITLE
Added alwaysRenderColumn

### PR DESCRIPTION
**Pull request by SatanEnglish**
Added a way to always render the columns even when they go out of horizontal bounds.

As discussed in #306. Example is included

Please check and advice about cleanUpAndRenderCells function as mentioned in discussion.

### CODE REVIEW

**Comment by SatanEnglish** [comment]
@ghiscoding Please tell me I've got it right this time

File: slick.grid.js [hunk 1]
```diff
@@ -3185,6 +3185,11 @@ if (typeof Slick === "undefined") {
         if (i <= options.frozenColumn) {
           continue;
         }
+        
+        // Ignore alwaysRenderedColumns
+        if (columns[i].alwaysRenderColumn){
```
Comment (ghiscoding) [inline slick.grid.js]: please also make sure the `columns[i]` exist before calling it's property. A bullet proof check would be 
`if (Array.isArray(columns) && columns[i] && columns[i].alwaysRenderColumn) {`

File: plugins/slick.rowdetailview.js [hunk 2]
```diff
@@ -572,6 +573,7 @@
         width: _options.width,
         resizable: false,
         sortable: false,
+        alwaysRenderColumn: _options.alwaysRenderColumn,
```
Comment (ghiscoding) [inline plugins/slick.rowdetailview.js]: so I assume that if user is not passing it, the ideal input should be `false`, then it's better to define it as such
`alwaysRenderColumn: _options.alwaysRenderColumn || false,`

**Comment by ghiscoding** [comment]
@SatanEnglish 
it's much better, I thought your Row Detail example was wrong but just realized that it's a completely new file. You're also touching the Core lib `slick.grid.js`, whenever you touch that file, you have to make sure your code is bullet proof since that could affect a lot of users and ideally test a few of the examples to make sure it's all good. I posted a couple of review comments for you to address

**Comment by SatanEnglish** [comment]
All good I'll have another look.
Think it should be bullet proof as I copied how to get the columns from a line in the same function.
But I can add more checks if your concerned.

As for _options.alwaysRenderColumn it's default is set in defaults for the plugin as is `true` this is the case that most people will want set. (But is setup so it can be overriden)

**Comment by ghiscoding** [comment]
Whenever we touch the core lib, I am (we are) very concerned. I got hit by this in the past and so did @6pac.

File: slick.grid.js (hunk 1 above)
Comment (SatanEnglish) [reply slick.grid.js]: Added this and checked my page still works

File: plugins/slick.rowdetailview.js (hunk 2 above)
Comment (SatanEnglish) [reply plugins/slick.rowdetailview.js]: This is in the defaults of the plugin and is set to `true` as would be main use case.

**Comment by SatanEnglish** [comment]
Added your bullet proof check. :)

**Comment by SatanEnglish** [comment]
Still concerned about `cleanUpAndRenderCells` function hoping for feedback as to if it needed a check in there.

**Comment by ghiscoding** [comment]
I posted more than 1 comment though, please see the full list of review

**Comment by SatanEnglish** [comment]
I did reply to both.

`alwaysRenderColumn: _options.alwaysRenderColumn || false`
Is not needed as it's set on the defaults of the plugin.

```
   var _defaults = {
      columnId: '_detail_selector',
      cssClass: 'detailView-toggle',
      expandedClass: null,
      collapsedClass: null,
      keyPrefix: '_',
      loadOnce: false,
      collapseAllOnSort: true,
      saveDetailViewOnScroll: true,
      useSimpleViewportCalc: false,
      alwaysRenderColumn: true, <------------------------
      toolTip: '',
      width: 30,
      maxRows: null
    }
```

**Comment by 6pac** [comment]
Happy to pull this as the core impact is minor. OK @ghiscoding ?
I will test after integrating but assume this has been tested already quite thoroughly.

**Comment by ghiscoding** [comment]
@6pac 
The code seems fine, I just didn't have a chance to test any of it locally though. I'll let you decide, if u want to merge right away

**Comment by ghiscoding** [comment]
@6pac 
I tried the PR with a few of the examples (about 10 various examples) and I don't see any problems.
So it looks good on from my perspective, I'm ok with merging it

Cheers mate

**Pull request merged**